docs/aws/audit/sgaudit/rules/ec2_public_instances_exist.mdx

```diff
@@ -23,6 +23,223 @@ HIPAA, NIST, SOC2
+   ```sh
+   aws ec2 describe-security-groups
+   ```
+
+2. **Revoke Inbound Rules Allowing Public Access:**
+   Revoke any inbound rules that allow public access (i.e., rules with a source of `0.0.0.0/0` for IPv4 or `::/0` for IPv6). For example, to revoke an inbound rule that allows SSH (port 22) from any IP address, use:
+   ```sh
+   aws ec2 revoke-security-group-ingress --group-id <security-group-id> --protocol tcp --port 22 --cidr 0.0.0.0/0
+   ```
+
+3. **Add Restrictive Inbound Rules:**
+   Add more restrictive inbound rules to allow access only from specific IP addresses or CIDR blocks. For example, to allow SSH access only from a specific IP address (e.g., `203.0.113.0`), use:
+   ```sh
+   aws ec2 authorize-security-group-ingress --group-id <security-group-id> --protocol tcp --port 22 --cidr 203.0.113.0/32
+   ```
+
+4. **Verify Security Group Rules:**
+   Verify that the security group rules have been updated correctly and no rules allow public access. You can describe the security group again to check the current rules:
+   ```sh
+   aws ec2 describe-security-groups --group-ids <security-group-id>
+   ```
+
+By following these steps, you can ensure that your EC2 instances are not publicly accessible through their associated security groups.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent EC2 instances from being publicly accessible in Security Groups using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. You can install it using pip if you haven't already:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session and create an EC2 client to interact with the AWS EC2 service.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+   ec2 = session.client('ec2')
+   ```
+
+3. **Describe Security Groups**:
+   Retrieve the security groups and their rules to identify any that allow public access (i.e., 0.0.0.0/0 or ::/0).
+   ```python
+   def get_public_security_groups(ec2):
+       public_sgs = []
+       response = ec2.describe_security_groups()
+       for sg in response['SecurityGroups']:
+           for permission in sg['IpPermissions']:
+               for ip_range in permission.get('IpRanges', []):
+                   if ip_range.get('CidrIp') == '0.0.0.0/0':
+                       public_sgs.append(sg['GroupId'])
+               for ipv6_range in permission.get('Ipv6Ranges', []):
+                   if ipv6_range.get('CidrIpv6') == '::/0':
+                       public_sgs.append(sg['GroupId'])
+       return public_sgs
+
+   public_sgs = get_public_security_groups(ec2)
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>